scripts/trade0x-kovan.js

```diff
@@ -0,0 +1,275 @@
+          takerFee: 'uint256',
+          expirationTimeSeconds: 'uint256',
+          salt: 'uint256',
+          makerAssetData: 'bytes',
+          takerAssetData: 'bytes',
+          makerFeeAssetData: 'bytes',
+          takerFeeAssetData: 'bytes',
+        },
+      },
+      'uint256[]',
+      'bytes[]',
+    ],
+    [taker, [signedOrder], [optionsToMint], [signedOrder.signature]],
+  )
+
+  const actionArgs = [
+    {
+      actionType: ActionType.OpenVault,
+      owner: taker,
+      secondAddress: ZERO_ADDR,
+      asset: ZERO_ADDR,
+      vaultId: vaultCounter.toString(),
+      amount: '0',
+      index: '0',
+      data: ZERO_ADDR,
+    },
+    {
+      actionType: ActionType.MintShortOption,
+      owner: taker,
+      secondAddress: taker,
+      asset: callOption1.address,
+      vaultId: vaultCounter.toString(),
+      amount: optionsToMint,
+      index: '0',
+      data: ZERO_ADDR,
+    },
+    {
+      actionType: ActionType.Call,
+      owner: taker,
+      secondAddress: trade0xCallee.address,
+      asset: ZERO_ADDR,
+      vaultId: vaultCounter.toString(),
+      amount: '0',
+      index: '0',
+      data: tradeCallData,
+    },
+    {
+      actionType: ActionType.DepositCollateral,
+      owner: taker,
+      secondAddress: payableProxyController.address,
+      asset: weth.address,
+      vaultId: vaultCounter.toString(),
+      amount: collateralToDeposit,
+      index: '0',
+      data: ZERO_ADDR,
+    },
+  ]
+
+  // pay protocol fee in ETH.
+  const gasPriceGWei = '50'
+  const gasPriceWei = web3.utils.toWei(gasPriceGWei, 'gwei')
+  // protocol require 70000 * gas price per fill
+  const feeAmount = new BigNumber(gasPriceWei).times(70000)
+  const operateValue = feeAmount.plus(new BigNumber(collateralToDeposit))
+
+  console.log("ETH needed: 💰 ", operateValue.toString())
+
+  const user1UsdcBalanceBefore = new BigNumber(await usdc.balanceOf(taker))
+  const marginPoolWethBalanceBefore = new BigNumber(await weth.balanceOf(marginPool.address))
+  const user1Call1BalanceBefore = new BigNumber(await callOption1.balanceOf(taker))
+  const oTokenSupplyBefore = new BigNumber(await callOption1.totalSupply())
+  const makerUsdcBalanceBefore = new BigNumber(await usdc.balanceOf(maker))
+  const makerCall1BalanceBefore = new BigNumber(await callOption1.balanceOf(maker))
+
+  console.log("MarginPool WETH balance before: 📊", marginPoolWethBalanceBefore.toString())
+  console.log("Maker USDC balance before: 📊", makerUsdcBalanceBefore.toString())
+  console.log("Maker Otoken balance before: 📊", makerCall1BalanceBefore.toString())
+  console.log("Taker USDC balance before: 📊", user1UsdcBalanceBefore.toString())
+  console.log("Taker Otoken balance before: 📊", user1Call1BalanceBefore.toString())
+  console.log("Otoken total supply before: 📊", oTokenSupplyBefore.toString())
+
+  console.log("Approving 0x Callee and calling Operate 🚀")
+
+  await callOption1.approve(trade0xCallee.address, optionsToMint, {from: taker})
```

Comment:
I thought we were going to use the permit callee for this approval?

Reply:
The Otoken in Kovan currently have no `permit()`, so I just used the `approve()`